libcob/intrinsic.c

```diff
@@ -5542,7 +5542,11 @@ cob_intr_random (const int params, ...)
 #else
 	} else if (rand_needs_seeding) {
 		/* first invocation without explicit seed, use a random one */
+#ifdef _WIN32
+		seed = (get_seconds_past_midnight () * (long)COB_MODULE_PTR) & 0xFFFFFFFF;
+#else
 		seed = get_seconds_past_midnight () * (long)COB_MODULE_PTR;
+#endif
```

Comment:
`Shouldn't this be `(unsigned long)COB_MODULE_PTR` as in the code a few lines above?

Reply:
Not sure it makes much difference, either way there will be an implicit signed-to-unsigned conversion involved at some point (`get_seconds_past_midnight` returns a signed `int`). If you prefer casting the pointer to `unsigned long` I'll change that.